[UG] Reformat find notes (#351)
AY2122S1-CS2103T-F13-3/tp: docs/UserGuide.md

## Quick Start

<div markdown="block" class="alert alert-info">

**:information_source: JDK Installation Guide**

You can install the required JDK and JRE from the
[Java SE Development Kit Downloads page](https://www.oracle.com/java/technologies/downloads/).

Here is a [website](https://docs.oracle.com/en/java/javase/17/install/overview-jdk-installation.html#GUID-8677A77F-231A-40F7-98B9-1FD0B48C346A)
to help you with the installation. Follow the guide for your operating system for detailed instructions.

> </div>

Reviewer: I thought Amy said just have to give the 2nd website and no need to be so long?

* Deletes the student at the specified `INDEX`.

Examples:
* `list` followed by `delete 2` deletes the 2nd student in TAB.

> * `find n/Betsy` followed by `ldelete 1 1` deletes the 1st lesson for the 1st student in the results of the `find` command.

Reviewer: This is `delete` not `ldelete` right?

Reviewer 2: i tot this is saying `find n/Betsy` filters the list of students then `ldelete` acts on the 1st student 's 1st lesson in filtered list

Reviewer: But this example is under the "Deleting a student" section though, maybe should go under "Find" section instead?

Reviewer 2: oh right, j saw a duplicate eg under `ldelete` section

* Deletes the student at the specified `INDEX`.

> * The index must be a valid index number of a student shown in the displayed list.

Reviewer: This can be inferred from line 227, I think

#### Finding students by fields: `find`

Finds all students whose fields match the given keyword(s), based on the specified find condition.

> You can find by tags, student fields and lesson fields.

Reviewer: Phrasing a bit strange here. Find what? Need some more time to think

> You can find by tags, student fields and lesson fields.

Reviewer 2: ```suggestion
You can find students by tags, student fields and lesson fields.
```
not sure if this sounds clearer

* You must specify at least one field to search for, and provide at least one keyword.<br>
  e.g. `find cond/any` or `find n/` are invalid commands.

##### Search by tags
* You can search by multiple tags. e.g. `t/paid t/new`
* One tag parameter can only have one keyword. e.g. `t/paid new` is invalid.
* The search is case-insensitive. e.g. keyword `new` will match `NEW`.
* Only whole words will be matched. e.g. keyword `paid` will not match `unpaid`

Example:
* `find t/unpaid t/New` will return all students with both of the tags `UNPAID` and `NEW`.

##### Search by other student fields
* You can specify one or more keywords for each parameter. e.g. `find n/Amad Ali`.
* A field needs to contain all specified keywords to be matched.<br>
  e.g. keywords `Amad Ali` will not match `Amad`, but it will match `Amad bin Ali`.

> * If you specify multiple of the same prefix, only the last prefix will be used for the search of that field.

Reviewer 2: ```suggestion
* If you specify multiple keywords with the same prefix, only the last prefix will be used to search that field.
```
Not sure if this sounds better

* A keyword can match a word partially. e.g. keyword `uni` will match `university`.

> * The search is case-insensitive.

Reviewer: Super minor thing but line 481,482 not in same order as line 470,471 ahaha

Author: what do you mean by this?

Reviewer: In 470,471, the case sensitivity point comes after how the word is matched

Example:
* `find n/John a/west coast` will return students with name that contains `John`, and address that contains `west` and `coast`.

##### Search by lesson fields
* Finds all students with at least one lesson that matches the lesson parameters.

<table id="find-param-table">
    <thead>
        <tr>
            <th style="text-align:center">Fields</th>
            <th style="text-align:center">Find behaviour</th>
        </tr>
    </thead>
    <tbody>
        <tr>

>             <td>Subject <br> Lesson Rate</td>

Reviewer: I thought we standardised to "Lesson rates" to match the prefixes?

            <td>
              <ul>
                <li>The same constraints for student fields apply.</li>
              </ul>
            </td>
        </tr>
        <tr>
            <td>Homework</td>
            <td>
              <ul>

>                 <li>Finds students with lessons that have at least one homework that matches all the keywords.</li>

Reviewer: Not sure about this phrasing/suggestion but the phrasing here seems a bit awkward

```suggestion
                <li>Finds students with lessons that have at least one piece of homework matching all the keywords.</li>
```

                <li>The same constraints for student fields apply.</li>
              </ul>
            </td>
        </tr>
        <tr>
            <td>Start Date</td>
            <td>
              <ul>

>                 <li>Accepts one keyword that must follow the Date format</li>

Reviewer: Link to what Date format we talking about?
As specified in [Managing Lessons](#managing-lessons) ?

                <li>Finds students with lessons that fall on the specified date.</li>
              </ul>
            </td>
        </tr>
        <tr>
            <td>Cancelled Date</td>
            <td>
              <ul>
                <li>Accepts one keyword that must follow the Date format</li>
                <li>Finds students with lessons that are cancelled on the specified date.</li>
              </ul>
            </td>
        </tr>
        <tr>
            <td>Time Range</td>
            <td>
              <ul>
                <li>Accepts one keyword that must follow the Time Range format</li>

>                 <li>Finds students with lessons that occur within the time range, excluding the start and end time.<br>

Reviewer: Is it within or overlapping? I'm not too sure about this

<div class="caption">Layout of a cancelled makeup lesson card.</div>

<div markdown="block" class="alert alert-info">
**:information_source: Note:**<br>
A lesson **must** have the following fields: a start date, a time range, a lesson rate and a subject.

A lesson can be identified by the index number shown in the lesson list of the student. The lesson index specified for commands must be a valid index number shown in the lesson list of the student.
</div>

<table id="lesson-param-table">
    <thead>
        <tr>
            <th style="text-align:center; padding: 10px">Category</th>
            <th style="text-align:center">Field</th>
            <th style="text-align:center">Prefix</th>
            <th style="text-align:center">Constraints</th>

>             <th style="text-align:center">Examples</th>

Reviewer: I'm a little confused.. you want to remove the table?

            <td>Recurrence Flag</td>
            <td><code>recurring/</code></td>
            <td><ul>
              <li>Indicates whether the lesson is recurring</li>
              <li>Optional parameter: end date of the recurrence</li>
            </ul></td>
            <td><ul>
              <li><code>recurring/</code></li>
              <li><code>recurring/30 Nov 2100</code></li>
            </ul></td>
        </tr>
        <tr>
            <td>Outstanding Fees</td>
            <td><code>f/</code></td>
            <td>Refer to <a href="#monetary-fields">Monetary fields</a> for more details.</td>
            <td><code>lvl/J1</code></td>
        </tr>
        <tr>
            <td>Homework</td>
            <td><code>hw/</code></td>
            <td>Maximum of 50 characters</td>
            <td><ul>
              <li><code>hw/Test 2</code> is valid</li>
              <li><code>hw/</code> clears <strong>all</strong> existing pieces of homework.</li>
            <li><code>hw/ hw/Worksheet 1</code> removes the existing pieces of homework and adds <code>Worksheet 1</code> to the homework set.</li>
            </ul></td>
        </tr>
    </tbody>
</table>

> * Additional fields of a lesson aside from those stated in the table are used for recording cancelled dates of a lesson. More details can be found in [Editing a lesson](#editing-a-lesson--ledit).<br>

Reviewer 2: why u delete this table tho?

* Subject
* Rate

For all monetary fields (lesson rates and outstanding fees), we follow the [Singapore convention](https://www.dfa.cornell.edu/treasurer/cash-management/processinginternational/intl-currency) of using a decimal point '.' to separate dollars and cents, The values of these fields should only contain numbers, with at most one decimal point, and two decimal places and should not start or end with a decimal point.
An optional field for both types of lesson is:
* Homework
* Outstanding Fees (refer to more details in this [section](#managing-lesson-fees))

</div>
Optional fields for a **recurring** lesson is:
* End date
* Cancelled dates

TAB has features to help to prevent scheduling mistakes.
For example, we help you avoid mistakes such as scheduling two lessons at the same slots by ensuring that lessons never overlap.
Additionally, we help to avoid accidentally scheduling lessons outside working hours (before 0800 and after 2200 hours).

In the future, we intend to add some nice-to-have features that will allow you to customise your working hours,
as well as specify minimum/maximum lesson durations so that accidents like mistyping a 10-minute lesson won't happen.

> Additionally, we will allow you to toggle these checks on and off, just in case you don't want them. You would also be able to customise your own recurrence rule for your recurring lessons.

Reviewer 2: why delete this?
